common/istio-1-14/istio-install/base/patches/disable-debugging.yaml

```diff
@@ -0,0 +1,18 @@
+# Pentest enahncement: check port 15010 & 8080 in istiod: According to https://istio.io/latest/docs/ops/best-practices/security/#control-plane port 15010
+# is not that problematic (only resource discovery). Other parts of the documentation also say| 15010 | GRPC | XDS and CA services (Plaintext, only for secure networks) | 
+# We have a secure network layer and only XDS is served. 
+# Port 8080 is not listed in the service and even if it would be somehow reachable by IP it only "offers read access". 
+# Nevertheless we set ENABLE_DEBUG_ON_HTTP=false do disable it entirely.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: istiod
+  namespace: istio-system
+spec:
+  template:
+    spec:
+      containers:
+      - name: discovery
+        env:
+        - name: ENABLE_DEBUG_ON_HTTP
```

Comment:
@juliusvonkohout does the `ENABLE_DEBUG_ON_HTTP` only affect the (deprecated) 8080 port, that has the `debugging interface` right?

Reply:
It does definitely not affect port 80 which serves the istio-ingressgateway and therefore centraldashboard. It was tested for several months on istio-cni. https://istio.io/latest/docs/ops/best-practices/security/#control-plane says "Port 8080 exposes the debug interface, which offers read access to a variety of details about the clusters state. This can be disabled by set the environment variable ENABLE_DEBUG_ON_HTTP=false on Istiod. Warning: many istioctl commands depend on this interface and will not function if it is disabled."